lib/BlockedBy.js

```diff
@@ -41,10 +41,13 @@ class BlockedBy extends NodeResque.Plugin {
 
         // If not blocking, but the same job is already waiting
         if (sameJobWaiting > 0) {
-            // get the first build that is waiting
-            let firstWaitingBuild = await this.queueObject.connection.redis.lindex(waitingKey, 0);
+            let waitingBuilds = await this.queueObject.connection.redis.lrange(waitingKey, 0, -1);
+
+            waitingBuilds = waitingBuilds.map(bId => parseInt(bId, 10));
+            waitingBuilds.sort((a, b) => a - b);
 
-            firstWaitingBuild = parseInt(firstWaitingBuild, 10);
+            // get the first build that is waiting
+            const firstWaitingBuild = waitingBuilds[0];
```

Comment:
Need to change this line to remove the match one instead of lpop: https://github.com/screwdriver-cd/queue-worker/blob/81640a61b3011907c25202635a83074ffbb642f5/lib/BlockedBy.js#L62

Reply:
👍  Making the change.